neqo-transport/src/connection/mod.rs

```diff
@@ -2282,7 +2269,8 @@ impl Connection {
             let aead_expansion = tx.expansion();
             builder.set_limit(profile.limit() - aead_expansion);
             builder.enable_padding(needs_padding);
-            debug_assert!(builder.limit() <= 2048);
+            let limit = builder.limit();
+            debug_assert!(limit <= 2048);
```

Comment:
Is this looking for side effects?

Reply:
Yes
```
error: do not call a function with mutable arguments inside of `debug_assert!`
    --> neqo-transport/src/connection/mod.rs:2285:27
     |
2285 |             debug_assert!(builder.limit() <= 2048);
     |                           ^^^^^^^^^^^^^^^
     |
     = help: for further information visit https://rust-lang.github.io/rust-clippy/master/index.html#debug_assert_with_mut_call
     = note: `-D clippy::debug-assert-with-mut-call` implied by `-D warnings`
     = help: to override `-D warnings` add `#[allow(clippy::debug_assert_with_mut_call)]`
```

Reply:
The fix you should use here is to change `PacketBuilder::limit` to:

```rust
    #[must_use]
    pub fn limit(&self) -> usize {
        self.limit
    }
```

... it is currently `&mut self` unnecessarily.

Reply:
FWIW, this is a great lint and I'm happy to see that it exists.